wal/wal.go

```diff
@@ -617,15 +617,23 @@ func (w *WAL) log(rec []byte, final bool) error {
 
 		// By definition when a record is split it means its size is bigger than
 		// the page boundary so the current page would be full and needs to be flushed.
-		// On contrary if we wrote a full record, we can fit more records of the batch
-		// into the page before flushing it.
-		if final || typ != recFull || w.page.full() {
+		// On contrary if we wrote a full record or last fragment of a record, we can
+		// fit more records of the batch into the page before flushing it.
+		if (typ != recFull && typ != recLast) || w.page.full() {
```

Comment:
I think this would be clearer to read as
```
if type == recFirst || type == recMiddle || w.page.full() {
...
}
```
then you can get rid of the entire "On contrary" sentence.

Reply:
@csmarchbanks Thanks for your comments, updated!